src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -55,9 +55,57 @@ namespace OpenLoco::World::MapGenerator
     // 0x004625D0
     static void generateLand(HeightMap& heightMap)
     {
-        _heightMap = heightMap.data();
-        call(0x004625D0);
-        _heightMap = nullptr;
+        for (auto pos : World::getDrawableTileRange())
+        {
+            const uint8_t q00 = heightMap[{ pos.x + 0, pos.y + 0 }];
+            const uint8_t q01 = heightMap[{ pos.x + 0, pos.y + 1 }];
+            const uint8_t q10 = heightMap[{ pos.x + 1, pos.y + 0 }];
+            const uint8_t q11 = heightMap[{ pos.x + 1, pos.y + 1 }];
+
+            const auto tile = TileManager::get(pos);
+            auto* surfaceElement = tile.surface();
+            if (surfaceElement == nullptr)
+            {
+                continue;
+            }
+
+            const uint8_t baseHeight = (q00 + q01 + q10 + q11) / 4;
```

Comment:
NB: vanilla is doing something closer to `std::max({q00, q01, q10, q11});`. However, results are substantially less smooth if I use that.